pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1650,3 +1660,97 @@ func protoEncodeFlowlogCounters(counters types.FlowlogCounters) *metrics.Flowlog
 func getDormantTime(ctx *zedagentContext) uint64 {
 	return uint64(ctx.globalConfig.GlobalValueInt(types.MetricInterval) * dormantTimeScaleFactor)
 }
+
+func getStorageDeviceMetrics(obj *types.ZDeviceMetrics) *zmet.StorageDeviceMetrics {
+	storageDevMetrics := new(metrics.StorageDeviceMetrics)
+	storageDevMetrics.Alloc = *proto.Uint64(obj.Alloc)
+	storageDevMetrics.Total = *proto.Uint64(obj.Space)
+	storageDevMetrics.DeflatedSpace = *proto.Uint64(obj.DSpace)
+	storageDevMetrics.ReplaceableSize = *proto.Uint64(obj.RSize)
+	storageDevMetrics.ExpandableSize = *proto.Uint64(obj.ESize)
+	storageDevMetrics.ReadErrors = *proto.Uint64(obj.ReadErrors)
+	storageDevMetrics.WriteErrors = *proto.Uint64(obj.WriteErrors)
+	storageDevMetrics.ChecksumErrors = *proto.Uint64(obj.ChecksumErrors)
+	storageDevMetrics.BytesRead = *proto.Uint64(obj.Bytes[types.ZIOTypeRead])
+	storageDevMetrics.BytesWrite = *proto.Uint64(obj.Bytes[types.ZIOTypeWrite])
+	storageDevMetrics.OpsCountRead = *proto.Uint64(obj.Ops[types.ZIOTypeRead])
+	storageDevMetrics.OpsCountWrite = *proto.Uint64(obj.Ops[types.ZIOTypeWrite])
+	storageDevMetrics.IOsInProgress = *proto.Uint64(obj.IOsInProgress)
+	storageDevMetrics.ReadTicks = *proto.Uint64(obj.ReadTicks)
+	storageDevMetrics.WriteTicks = *proto.Uint64(obj.WriteTicks)
+	storageDevMetrics.IOsTotalTicks = *proto.Uint64(obj.IOsTotalTicks)
+	storageDevMetrics.WeightedIOTicks = *proto.Uint64(obj.WeightedIOTicks)
+	return storageDevMetrics
+}
+
+func fillStorageZVolMetrics(zvol *types.StorageZVolMetrics) *metrics.StorageZVolMetric {
```

Comment:
What's the naming scheme here? You are introducing two new functions to set/get/fill fields in the API messages, yet you use different prefixes for the two functions.

Reply:
Thanks. Corrected.

At the pkg/pillar/cmd/zedagent/handlemetrics.go level where the message is filled before sending 
now all functions are prefixed with fill...

At the pkg/pillar/zfs/zfs.go level where we collect data from zfs, the functions are prefixed with Get...